tests/paramgrill.c

```diff
@@ -1245,80 +1381,31 @@ static int sanitizeVarArray(varInds_t* varNew, const int varLength, const varInd
 }
 
 /* res should be NUM_PARAMS size */
-/* constructs varArray from ZSTD_compressionParameters style parameter */
-static int variableParams(const ZSTD_compressionParameters paramConstraints, varInds_t* res) {
+/* constructs varArray from paramValues_t style parameter */
+/* pass in using dict. */
+static int variableParams(const paramValues_t paramConstraints, varInds_t* res, const int usingDictionary) {
+    varInds_t i;
     int j = 0;
-    if(!paramConstraints.windowLog) {
-        res[j] = wlog_ind;
-        j++;
-    }
-    if(!paramConstraints.chainLog) {
-        res[j] = clog_ind;
-        j++;
-    }
-    if(!paramConstraints.hashLog) {
-        res[j] = hlog_ind;
-        j++;
-    }
-    if(!paramConstraints.searchLog) {
-        res[j] = slog_ind;
-        j++;
-    }
-    if(!paramConstraints.searchLength) {
-        res[j] = slen_ind;
-        j++;
-    }
-    if(!paramConstraints.targetLength) {
-        res[j] = tlen_ind;
-        j++;
-    }
-    if(!paramConstraints.strategy) {
-        res[j] = strt_ind;
-        j++;
-    }
-    return j;
-}
-
-/* bin-search on tlen_table for correct index. */
-static int tlen_inv(U32 x) {
-    int lo = 0;
-    int hi = TLEN_RANGE;
-    while(lo < hi) {
-        int mid = (lo + hi) / 2;
-        if(tlen_table[mid] < x) {
-            lo = mid + 1;
-        } if(tlen_table[mid] == x) {
-            return mid;
-        } else {
-            hi = mid;
+    for(i = 0; i < NUM_PARAMS; i++) {
+        if(paramConstraints.vals[i] == PARAM_UNSET) {
+            if(i == fadt_ind && !usingDictionary) continue; /* don't use fadt if no dictionary */
+            res[j] = i; j++;
         }
     }
-    return lo;
+    return j;
 }
 
 /* amt will probably always be \pm 1? */
 /* slight change from old paramVariation, targetLength can only take on powers of 2 now (999 ~= 1024?) */
 /* take max/min bounds into account as well? */
-static void paramVaryOnce(const varInds_t paramIndex, const int amt, ZSTD_compressionParameters* ptr) {
-    switch(paramIndex)
-    {
-        case wlog_ind: ptr->windowLog    += amt; break;
-        case clog_ind: ptr->chainLog     += amt; break;
-        case hlog_ind: ptr->hashLog      += amt; break;
-        case slog_ind: ptr->searchLog    += amt; break;
-        case slen_ind: ptr->searchLength += amt; break;
-        case tlen_ind: 
-            ptr->targetLength = tlen_table[MAX(0, MIN(TLEN_RANGE - 1, tlen_inv(ptr->targetLength) + amt))];
-            break;
-        case strt_ind: ptr->strategy     += amt; break;
-        default: break;
-    }
+static void paramVaryOnce(const varInds_t paramIndex, const int amt, paramValues_t* ptr) {
+    ptr->vals[paramIndex] = rangeMap(paramIndex, invRangeMap(paramIndex, ptr->vals[paramIndex]) + amt); //TODO: bounds check. 
```

Comment:
which bound check ?

Reply:
It was added in invRangeMap + rangeMap, I just forgot to remove this comment